src/cmap/connection_pool.ts

```diff
@@ -391,10 +399,10 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
       this[kConnections].unshift(connection);
     }
 
-    this[kCheckedOut].delete(connection);
+    const wasConnectionDeleted = this[kCheckedOut].delete(connection);
```

Comment:
we can remove this gate now that we have the guard in L390

Reply:
done